dask/core.py

```diff
@@ -1,10 +1,24 @@
 from collections import defaultdict
+from contextlib import contextmanager
 
+from .config import get as config_get, set as config_set
 from .utils_test import add, inc  # noqa: F401
 
 no_default = "__no_default__"
 
 
+@contextmanager
+def annotate(**kwargs):
```

Comment:
I think that we could us a docstring here.  My *guess* is that this will be the most commonly read piece of information about annotations.  https://docs.dask.org/en/latest/develop.html#docstrings

In particular, I suspect that it will be useful to include a couple of common examples, perhaps with special values from dask.distributed like `priority` and retries`

Reply:
Also, I recommend that we move this to base.py.  I think that core.py is mostly based around direct graph manipulation.

Reply:
This might also allow us to do `from . import config`, which might allow us to avoid the underscore namespacing.

Reply:
Moved to base.py

Reply:
I've added some documentation. More could be added when HLG transmission to the scheduler is implemented, especially in relation to SchedulerPlugins. It would probably be good to mention the current annotation taxonomy understood by the distributed scheduler (priority, retry, worker, resource).

I've refrained at this point because this isn't yet functional. What do you think?